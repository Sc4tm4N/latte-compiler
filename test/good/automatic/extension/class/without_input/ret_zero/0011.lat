class Node {
  string a;

  void printMe() {
    printString(a);
  }

  void setMe1(string b) {
    self.a = self.a + a + b;
  }

  void setMe2(string a) {
    self.a = self.a + a;
  }

  string getMe() {
    return a;
  }

  string extendMe(string suffix) {
    return a + suffix;
  }
}

int main() {
  Node node = new Node;
  node.a = "kappa";
  printString(node.getMe());
  printString(node.a);
  node.printMe();


  node.setMe1(" kapcie");
  printString(node.getMe());
  printString(node.a);
  node.printMe();

  node.setMe2(" z");
  printString(node.getMe());
  printString(node.a);
  node.printMe();

  node.setMe2(" biedry, skrrrt!");
  printString(node.getMe());
  printString(node.a);
  node.printMe();

  string s = node.extendMe("SUFFIX!");
  printString(node.extendMe(s + "x2"));

  printString(node.getMe());
  printString(node.a);
  node.printMe();
  return 0;
}
