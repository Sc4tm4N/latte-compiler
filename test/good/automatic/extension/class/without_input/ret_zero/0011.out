kappa
kappa
kappa
kappakappa kapcie
kappakappa kapcie
kappakappa kapcie
kappakappa kapcie z
kappakappa kapcie z
kappakappa kapcie z
kappakappa kapcie z biedry, skrrrt!
kappakappa kapcie z biedry, skrrrt!
kappakappa kapcie z biedry, skrrrt!
kappakappa kapcie z biedry, skrrrt!kappakappa kapcie z biedry, skrrrt!SUFFIX!x2
kappakappa kapcie z biedry, skrrrt!
kappakappa kapcie z biedry, skrrrt!
kappakappa kapcie z biedry, skrrrt!
